stress_test.py

```diff
@@ -0,0 +1,490 @@
+- Many runs that use all sorts of large Docker images
+- Call CodaLab repeatedly to get information to check if the system is still responsive
+"""
+
+# List of random, large docker images
+_LARGE_DOCKER_IMAGES = [
+    'adreeve/python-numpy',
+    'larger/rdp:dev',
+    'openjdk:11.0.5-jre',
+    'tensorflow/tensorflow:devel-gpu',
+    'iwane/numpy-matplotlib',
+    'kelvinxu/baselines',
+    'aaronyalai/openaibaselines:gym3',
+    'couchbase:latest',
+    'large64/docker-test:ruby',
+    'pytorch/pytorch:1.3-cuda10.1-cudnn7-devel',
+]
+
+
+class TestFile:
+    """
+    Wrapper around temporary file. Creates the file at initialization.
+
+    Args:
+        file_name (str): Name of the file to create and manage
+        size_mb (int): Size of the test file in megabytes
+        content (str): Content to write out to the file. Default is None.
+    """
+
+    def __init__(self, file_name, size_mb=1, content=None):
+        self._file_name = file_name
+        if content == None:
+            self._size_mb = size_mb
+            self._make_random_file()
+        else:
+            self._write_content(content)
+
+    def _make_random_file(self):
+        with open(self._file_name, 'wb') as file:
+            file.seek(self._size_mb * 1024000)  # seek takes in file size in terms of bytes
+            file.write(b'0')
+        print('Created file {} of size {} MB.'.format(self._file_name, self._size_mb))
+
+    def _write_content(self, content):
+        with open(self._file_name, 'w') as file:
+            file.write(content)
+
+    def name(self):
+        return self._file_name
+
+    def delete(self):
+        '''
+        Removes the file.
+        '''
+        if os.path.exists(self._file_name):
+            os.remove(self._file_name)
+            print('Deleted file {}.'.format(self._file_name))
+        else:
+            print('File {} has already been deleted.'.format(self._file_name))
+
+
+class StressTestRunner(ABC):
+    """
+    Abstract class that holds common logic for stress test runners.
+
+    Args:
+        cl: cl instance used to run CodaLab commands
+    """
+
+    def __init__(self, cl):
+        self._cl = cl
+
+    def run(self):
+        print('Running stress tests...')
+        self._start_heartbeat()
+        self._test_large_bundle()
+        self._test_many_bundle_uploads()
+        self._test_many_worksheet_copies()
+        self._test_parallel_runs()
+        self._test_many_docker_runs()
+        self._test_infinite_memory()
+        self._test_infinite_gpu()
+        self._test_infinite_disk()
+        self._test_many_disk_writes()
+        print('Done.')
+
+    def _start_heartbeat(self):
+        # Start heartbeats in the background. Each heartbeat creates a worksheet and prints its content.
+        t = Thread(target=self._heartbeat)
+        t.daemon = True
+        t.start()
+
+    def _heartbeat(self):
+        while True:
+            # Create a worksheet in a separate thread and check if it times out or not.
```

Comment:
If you just want a heartbeat, why not choose something that doesn't mutate?

Reply:
I replaced it with a read-only query instead.